pkg/controllers/status/work_status_controller.go

```diff
@@ -202,85 +194,18 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
 		return nil
 	}
 
-	desiredObj, err := c.getRawManifest(workObject.Spec.Workload.Manifests, observedObj)
-	if err != nil {
-		return err
-	}
-
-	clusterName, err := names.GetClusterName(workNamespace)
-	if err != nil {
-		klog.Errorf("Failed to get member cluster name: %v", err)
-		return err
-	}
-
-	// we should check if the observed status is consistent with the declaration to prevent accidental changes made
-	// in member clusters.
-	needUpdate, err := c.ObjectWatcher.NeedsUpdate(clusterName, desiredObj, observedObj)
-	if err != nil {
+	if !helper.IsResourceApplied(&workObject.Status) {
+		err = fmt.Errorf("work hadn't been applied yet")
+		klog.Errorf("Failed to reflect status of %s(%s/%s) to Work(%s/%s): %v.",
+			observedObj.GetKind(), observedObj.GetNamespace(), observedObj.GetName(), workNamespace, workName, err)
 		return err
 	}
 
-	if needUpdate {
```

Comment:
How is this part implemented in execution controller?

Reply:
needUpdate is decided by `c.ObjectWatcher.NeedsUpdate(clusterName, desiredObj, observedObj)`, and it checks the labels when the generation is the same, but `GenerationChangedPredicate` only checks the generation, is this expected?

Reply:
Like this:  
https://github.com/karmada-io/karmada/pull/3808/files/cdbc943859f432dd125ca24c4fe2e3d7e468ae2e#r1267840563  

I will think more about this. 

Nice remind, thanks.